### PR TITLE
DownloadStream

### PR DESCRIPTION
**Pull request by marek-vysoky**
Changing implementation for DownloadStreamAsync, adding unit test DownloadStream_ValidResponse_ValidStream

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @marek-vysoky has signed the CLA. Thanks for the pull request!

**Comment by coolboy** [comment]
@marek-vysoky conflict

**Comment by marek-vysoky** [comment]
I've merged master to this branch, resolved the conflict and pushed the changes

File: Box.V2/Managers/BoxFilesManager.cs [hunk 1]
```diff
@@ -48,19 +48,19 @@ public BoxFilesManager(IBoxConfig config, IBoxService service, IBoxConverter con
...
-
+            var uri = await GetDownloadUriAsync(id, versionId);
+            BoxRequest request = new BoxRequest(uri)
+            {
+                Timeout = timeout
+            };
             IBoxResponse<Stream> response = await ToResponseAsync<Stream>(request).ConfigureAwait(false);
-
             return response.ResponseObject;
+
```
Comment (coolboy) [inline Box.V2/Managers/BoxFilesManager.cs]: remove this empty line

File: Box.V2/Managers/BoxFilesManager.cs [hunk 2]
```diff
@@ -48,19 +48,19 @@ public BoxFilesManager(IBoxConfig config, IBoxService service, IBoxConverter con
...
+        /// <param name="timeout">Optional timeout for response</param>
         /// <returns>MemoryStream of the requested file</returns>
         public async Task<Stream> DownloadStreamAsync(string id, string versionId = null, TimeSpan? timeout = null)
         {
-            id.ThrowIfNullOrWhiteSpace("id");
-
-            BoxRequest request = new BoxRequest(_config.FilesEndpointUri, string.Format(Constants.ContentPathString, id)) { Timeout = timeout }
-                .Param("version", versionId);
-
+            var uri = await GetDownloadUriAsync(id, versionId);
```
Comment (coolboy) [inline Box.V2/Managers/BoxFilesManager.cs]: why removed the check for the id.ThrowIfNullOrWhiteSpace("id")

Comment (marek-vysoky) [inline Box.V2/Managers/BoxFilesManager.cs]: I've removed it because GetDownloadUriAsync has first validation id.ThrowIfNullOrWhiteSpace("id");

**Comment by marek-vysoky** [comment]
I've removed empty line after return.

**Comment by cburnette** [comment]
@coolboy I've reverted this back to not use GetDownloadUriAsync.  I get what he was thinking, but it's going to be a bit more performant to just let the HTTP engine follow the redirect automatically rather that make an api call via GetDownloadUriAsync and then make another call to download.

**Pull request merged**